packages/replay/src/replay.ts

```diff
@@ -1071,7 +1054,9 @@ export class ReplayContainer implements ReplayContainerInterface {
    * Should never be called directly, only by `flush`
    */
   private async _runFlush(): Promise<void> {
-    if (!this.session || !this.eventBuffer) {
+    const replayId = this.getSessionId();
```

Comment:
capturing the replayId to use at the very top here. With this, _if_ this runs out of sync while processing, we should still never send data to the _wrong_ replay ID, at least.